File: 0001-two-sum/0001-two-sum.java
import java.util.HashMap;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap <Integer, Integer> twoSum = new HashMap<Integer, Integer>();

        int x = 0;

        for(int i =0;i<nums.length;i++)
        {
            x = target - nums[i];
            if(twoSum.containsKey(x))
            {
                return new int[] {twoSum.get(x),i};
            }
            else
            {
                twoSum.put(nums[i],i);
            }

        }  
        return new int[] {};      
    }
    
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        
        for i, n in enumerate(nums):
            diff = target - n
            
            if diff in map:
                return [i, map[diff]]
            map[n] = i
        return []
        
        